src/backend/kvm/mod.rs

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use super::Loader;
+use crate::backend::kvm::data::{dev_kvm, kvm_version};
+use crate::backend::kvm::mem::Region;
+use anyhow::Result;
+use kvm_bindings::bindings::kvm_userspace_memory_region;
+pub use kvm_bindings::kvm_userspace_memory_region as KvmUserspaceMemoryRegion;
+use kvm_ioctls::Kvm;
+use kvm_ioctls::{VcpuFd, VmFd};
+use mmarinus::{perms, Map};
+use std::collections::VecDeque;
+use std::sync::Arc;
+use x86_64::VirtAddr;
+
+mod builder;
+mod config;
+mod data;
+mod mem;
+mod thread;
+
+impl Keep {
+    pub fn map(&mut self, pages: Map<perms::ReadWrite>, to: usize) -> std::io::Result<&mut Region> {
+        let region = kvm_userspace_memory_region {
+            slot: self.regions.len() as u32,
+            flags: 0,
+            guest_phys_addr: to as u64,
+            memory_size: pages.len() as u64,
+            userspace_addr: pages.addr() as u64,
+        };
+
+        unsafe { self.vm_fd.set_user_memory_region(region)? };
+
+        self.regions.push(Region::new(region, pages));
+        Ok(self.regions.last_mut().unwrap())
+    }
+}
+
+struct Keep {
+    kvm_fd: Kvm,
+    vm_fd: VmFd,
+    cpu_fds: VecDeque<VcpuFd>,
```

Comment:
Why is this a `VecDeque`? You always push and pop from the front, which is the same as `Vec`, no?

Reply:
hmm, right